File: Perl/abuild-1.1.6/doc/example/basic/java-library/src/java/com/example/basic/BasicLibrary.java
package com.example.basic;

public class BasicLibrary
{
    private int n;

    public BasicLibrary(int n)
    {
	this.n = n;
    }

    public void hello()
    {
	System.out.println("Hello.  This is BasicLibrary(" + n + ").");
    }
}

File: Perl/abuild-1.1.6/doc/example/basic/cxx-library/BasicLibrary.cc
#include "BasicLibrary.hh"
#include <iostream>

BasicLibrary::BasicLibrary(int n) :
    n(n)
{
}

void
BasicLibrary::hello()
{
    std::cout << "Hello.  This is BasicLibrary(" << n << ")." << std::endl;
}
